app/submitter/convert_payload_0_0_1.py

```diff
@@ -37,7 +38,11 @@ def convert_answers_to_payload_0_0_1(
                 metadata,
                 answer_store,
                 list_store,
-                current_location=block_id,
+                current_location=Location(
+                    block_id=block_id,
+                    section_id=routing_path.section_id,
+                    list_item_id=routing_path.list_item_id,
+                ),
```

Comment:
We don't tend to initialise variables in calls to other methods, `current_location` can be initialised outside like `block` is.